yomigeek/mtracker-app
# #157753217 User Login API

#### What does this PR do?
* Create an API endpoint to for a user to login with required credentials

#### Description of Task to be completed?
* User should be able to login through this endpoint

#### How should this be manually tested?
* Clone the project to your local repository from git clone https://github.com/yomigeek/mtracker-app.git
* All necessary dependencies as stated in the README.md file should be installed
* run $ npm install
* Run app locally using $ npm run start
* Open the postman application and enter the URL http://localhost:5000/api/v1/auth/login with POST request method selected and also enter the required fields such as email, password
```
{
	"email":  "user@gmail.com",
	"password": "11111"
}
```
* After the above is successful, you should get the following response: 
```
{
    "status": "success",
    "message": "Login successful",
    "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOjIsInVzZXJSb2xlIjoidXNlciIsImlhdCI6MTUyNzE1NzM4MSwiZXhwIjoxNTI3MzMwMTgxfQ.wBCHqUMxfVYMmH4EEU-rRMZUsF_cBhIuQ8IRWLV2IJs"
}
```
#### Any background context you want to provide?
All  fields are important i.e email and password must be supplied

####  What are the relevant pivotal tracker stories?
story type: feature
story id: 157753217

#### What are the relevant Github Issues?
None

#### Questions:
None

![login](https://user-images.githubusercontent.com/5827585/40490707-2f174454-5f64-11e8-898e-5483942b72d4.PNG)

![login2](https://user-images.githubusercontent.com/5827585/40490935-c23e93b8-5f64-11e8-80e2-20da4529635e.PNG)
`
